### PR TITLE
feat: add updateStrategy settings for ingress and dashboard

### PR DESCRIPTION
**Pull request by ikatlinsky**
As per https://github.com/apache/apisix-helm-chart/issues/582

### CODE REVIEW

File: charts/apisix-dashboard/values.yaml [hunk 1]
```diff
@@ -165,6 +165,10 @@ autoscaling:
   targetCPUUtilizationPercentage: 80
   # targetMemoryUtilizationPercentage: 80
 
+# -- Update strategy for apisix deployment
```
Comment (Gallardot) [inline charts/apisix-dashboard/values.yaml:168]: typo

File: charts/apisix-ingress-controller/values.yaml [hunk 2]
```diff
@@ -160,6 +160,10 @@ autoscaling:
   targetCPUUtilizationPercentage: 80
   # targetMemoryUtilizationPercentage: 80
 
+# -- Update strategy for apisix deployment
```
Comment (Gallardot) [inline charts/apisix-ingress-controller/values.yaml:163]: typo

**Review by Gallardot — CHANGES_REQUESTED** [review]
Rest LGTM

**Review by Gallardot — APPROVED** [review]

**Review by tao12345666333 — COMMENTED** [review]
You don't change the APISIX's real configuration?

**Comment by ikatlinsky** [comment]
What do you mean? My idea was to allow users to modify the default update strategy for deployments. Currently, all pods are  defaulted to rollout update with maxSurge and maxUnavailable = 25%.

**Comment by ikatlinsky** [comment]
@tao12345666333 not sure if I answered your question, make me know if you are interested in another aspect of the pr

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Pull request merged**